### DISCUSSION TITLE
Migrate Typescript, auto import js file problem in VScode.

### DISCUSSION THREAD

**Discussion by thienscmon** [vercel/next.js#67408] [Help]
### Summary

My project was using `js` and was converting to `typescript`. Nextjs detected and reconfigured the tsconfig.json file for me. The problem I'm having is that auto import from js files to tsx/ts files doesn't work unless I add them to `include` in `tsconfig.json` like this:
```
 "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"]
```

I suddenly realized that this would cause `next build` to perform type checking errors on the js files, causing the build to fail.
Is there a way to fix auto import without including `.js/jsx` inside tsconfig?
Sorry for my bad English

My tsconfig file
```
{
  "compilerOptions": {
    "lib": ["esnext", "dom"],
    "paths": {
      "@/*": ["./*"]
    },
    "baseUrl": ".",
    "checkJs": true,
    "strict": true,
    "jsx": "preserve",
    "moduleResolution": "node",
    "allowJs": true,
    "skipLibCheck": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "resolveJsonModule": true,
    "isolatedModules": true
  },
  "exclude": ["node_modules", "dist"],
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"]
}
```

**Comment by SouthLink** [comment]
You can use this configuration to tell next to ignore TS errors in the build
// next.config.js
```
const nextConfig = {
 typescript: {
    ignoreBuildErrors: true,
  }
 }

module.exports = nextConfig;
```
